File: python/astnode.py
class ASTNode:
    def __init__(self, op, left, mid, right, value):
        self.op = op
        self.left = left
        self.mid = mid
        self.right = right
        self.value = value
    
    @staticmethod
    def mkAstLeaf(op, value):
        return ASTNode(op, None, None, None, value)
    
    @staticmethod
    def mkAstUnary(op, left, value):
        return ASTNode(op, left, None, None, value)

File: julia/src/astnode.jl
mutable struct ASTNode
    op::ASTNodeOp
    left::Union{ASTNode, Nothing}
    right::Union{ASTNode, Nothing}
    value::Union{Int64, LLVM.Value}

    function ASTNode(op, left, right, intValue)
        new(op, left, right, intValue)
    end

    function ASTNode(op, intValue)
        new(op, nothing, nothing, intValue)
    end

    function ASTNode(op, left, intValue)
        new(op, left, nothing, intValue)
    end
end